stacker/hooks/aws_lambda.py

```diff
@@ -81,8 +81,9 @@ def _calculate_hash(files, root):
     file_hashes = []
     for fname in files:
         f = os.path.join(root, fname)
-        with open(f) as fd:
-            file_hashes.append(hashlib.md5(fd.read()).hexdigest())
+        with open(f, "rb") as fd:
+            file_hashes.append("%s:%s" % (fname,
+                                          hashlib.md5(fd.read()).hexdigest()))
```

Comment:
One last thing (sorry that I forgot to mention earlier): it's probably a bit more efficient to use a single hash object and incrementally append to it. That makes it easy to use `\0` to separate path and content, which is better as it is almost always disallowed in filenames, and not read the whole file to memory.

```python
hash = hashlib.md5()
for fname in files:
    f = os.path.join(root, fname)
    with open(f, "rb") as fd:
        hash.update(fname + "\0")
        for chunk in iter(lambda: f.read(4096), ''):
            hash.update(chunk)

return hash.hexdigest()